### PR TITLE
fix: keep fused_softcap finite on saturation-range logits (PRPUNDIT-5)

### PR DESCRIPTION
**Pull request by jiagaoxiang**
## Summary
- Fixes `fused_softcap_kernel` overflow: compute tanh in FP32 and clamp `|x|` before `exp(2x)` so finite logits such as 2000 with softcapping 30 saturate near ±30 instead of storing NaN.
- Adds an AMD GPU unit test that reaches the kernel through the public `fused_softcap` wrapper for FP32 and BF16, including a length not divisible by `BLOCK_SIZE=1024`, and compares against a float64 scaled-tanh reference.

Closes test gap **PRPUNDIT-5**.

## Test plan
- [x] `pytest tests/unit_tests/backends/megatron/test_logits_processor.py` on MI325X (FP32 and BF16)

### CODE REVIEW

File: primus/backends/megatron/core/extensions/logits_processor.py [hunk 1]
```diff
@@ -25,13 +25,15 @@ def fused_softcap_kernel(
     # Load values
     x = tl.load(full_logits_ptr + offsets, mask=mask)
 
-    # Perform operations in-place
+    # tanh saturates well before |x|~9. Clamping 2x keeps tl.exp finite in FP32
+    # (a logit of 2000 with softcapping 30 used to overflow exp(2x) to inf and
+    # store NaN instead of a value near +30). Compute in FP32 so BF16 inputs
+    # follow the same saturation contract.
```
[the remark this answers is not in the document]
Comment (jiagaoxiang) [reply primus/backends/megatron/core/extensions/logits_processor.py]: Accepted. The comment now says we clamp the scaled `x` so `exp(2x)` stays finite, which matches the code (`clamp(x)` then `exp(2.0 * x)`).

**Review by Xiaoming-AMD — APPROVED** [review]
